map.go

```diff
@@ -274,6 +274,20 @@ func (m *Map) Lookup(key, valueOut interface{}) error {
 	}
 }
 
+// LookupAndDelete retrieves and deletes a value from a Map.
+func (m *Map) LookupAndDelete(key, valueOut interface{}) error {
+	valueBytes := make([]byte, m.fullValueSize)
```

Comment:
You need to use makeBuffer here. The function currently doesn't change `valueOut`.